daemon/volumes.go

```diff
@@ -205,11 +236,22 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 		}
 
 		if len(bind.Name) > 0 && len(bind.Driver) > 0 {
+			// set the label
+			if err := label.SetFileCreateLabel(container.MountLabel); err != nil {
+				return fmt.Errorf("Unable to setup default labeling for volume creation %s: %v", bind.Source, err)
+			}
+
+			// create the volume
 			v, err := createVolume(bind.Name, bind.Driver)
 			if err != nil {
 				return err
```

Comment:
On 05/26/2015 05:41 PM, Arnaud Porterie wrote:
>
> In daemon/volumes.go
> <https://github.com/docker/docker/pull/12572#discussion_r31082773>:
>
> > @@ -205,11 +236,22 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
> >  		}
> >  
> >  		if len(bind.Name) > 0 && len(bind.Driver) > 0 {
> > +			// set the label
> > +			if err := label.SetFileCreateLabel(container.MountLabel); err != nil {
> > +				return fmt.Errorf("Unable to setup default labeling for volume creation %s: %v", bind.Source, err)
> > +			}
> > +
> > +			// create the volume
> >  			v, err := createVolume(bind.Name, bind.Driver)
> >  			if err != nil {
> >  				return err
>
> How bad would it be to return on this error case without the label
> being reset? (cc @rhatdan <https://github.com/rhatdan>)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/12572/files#r31082773>.
>
It could cause problems. Basically the call to SetFileCreateLabel(LABEL)

Tells the kernel that all file objects created by this process thread
should be labeled LABEL.  Until it is reset to "", which tells the
kernel to go back to the default.  We should defer setting the greate
label to "" which is what my code originally did.